src/custom/state/orders/utils.ts

```diff
@@ -227,14 +258,19 @@ export function getEstimatedExecutionPrice(
     return limitPrice
   }
 
-  const amountMinusFees = inputAmount.subtract(feeAmount)
+  // Check what's left to sell, discounting the surplus, if any
+  const { sellAmount } = getRemainderAmountsWithoutSurplus(order)
+  const remainingSellAmount = CurrencyAmount.fromRawAmount(order.inputToken, sellAmount)
+
+  // Use the remaining sell amount for the calculations, as the fee will be based on that
+  const amountMinusFees = remainingSellAmount.subtract(feeAmount)
```

Comment:
This is the "meat" of the change here.
Now using the remaining amount of the order for calculating the estimated execution price